crates/fluvio-spu/src/core/global_context.rs

```diff
@@ -197,17 +202,24 @@ mod file_replica {
                     "old follower replica exists, promoting to leader"
                 );
 
-                self.leaders_state()
+                if let Err(err) = self
+                    .leaders_state()
                     .promote_follower(
                         self.config().into(),
                         follower_replica,
                         new_replica.clone(),
                         self.status_update_owned(),
+                        self,
                     )
-                    .await;
+                    .await
+                {
+                    error!("follower promotion failed: {err}");
+                    outputs.push(ReplicaChange::StorageError(err));
+                };
             } else {
                 error!("follower replica {} didn't exists!", old_replica.id);
             }
+            outputs
         }
 
         pub async fn apply_replica_update(
```

Comment:
Is these formatting changes only?  If so, would prefer to move out of this PR since it's not related to this feature

Reply:
This is not formatting. This is propagating errors up from replica initialization.